src/sentry/static/sentry/app/views/onboarding/integrationSetup.tsx

```diff
@@ -0,0 +1,390 @@
+};
+
+const recordAnalyticsDocsClicked = ({organization, project, platform}: AnalyticsOpts) =>
+  analytics('onboarding_v2.full_docs_clicked', {
+    org_id: organization.id,
+    project: project?.slug,
+    platform,
+  });
+
+type Props = StepProps & {
+  api: Client;
+  organization: Organization;
+  integrationSlug: string;
+};
+
+type State = {
+  loadedPlatform: PlatformKey | null;
+  hasError: boolean;
+  provider: IntegrationProvider | null;
+  installed: boolean;
+};
+
+class IntegrationSetup extends React.Component<Props, State> {
+  state: State = {
+    loadedPlatform: null,
+    hasError: false,
+    provider: null,
+    installed: false,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(nextProps: Props) {
+    if (
+      nextProps.platform !== this.props.platform ||
+      nextProps.project !== this.props.project
+    ) {
+      this.fetchData();
+    }
+  }
+
+  get manualSetupUrl() {
+    const {search} = window.location;
+    // honor any existing query params
+    const separator = search.includes('?') ? '&' : '?';
+    return `${search}${separator}manual=1`;
+  }
+
+  get platformDocs() {
+    // TODO: make dynamic based on the integration
+    return 'https://docs.sentry.io/product/integrations/aws-lambda/';
+  }
+
+  fetchData = async () => {
+    const {api, organization, platform, integrationSlug} = this.props;
+
+    if (!integrationSlug) {
+      return;
+    }
+
+    try {
+      const endpoint = `/organizations/${organization.slug}/config/integrations/?provider_key=${integrationSlug}`;
+      const integrations = await api.requestPromise(endpoint);
+      const provider = integrations.providers[0];
+
+      this.setState({provider, loadedPlatform: platform, hasError: false});
+    } catch (error) {
+      this.setState({hasError: error});
+      throw error;
+    }
+  };
+
+  handleFullDocsClick = () => {
+    const {organization, project, platform} = this.props;
+    recordAnalyticsDocsClicked({organization, project, platform});
+  };
+
+  handleAddIntegration = () => {
+    this.setState({installed: true});
+  };
+
+  renderIntegrationInstructions() {
+    const {organization, platform} = this.props;
+    const {loadedPlatform, provider} = this.state;
+    if (!provider) {
+      return null;
+    }
+
+    const currentPlatform = loadedPlatform ?? platform ?? 'other';
+
+    return (
+      <React.Fragment>
+        <TitleContainer>
+          <StepHeading step={2}>
+            {t(
+              'Automatically instrument %s',
+              platforms.find(p => p.id === currentPlatform)?.name ?? ''
+            )}
+          </StepHeading>
+          <motion.div
+            variants={{
+              initial: {opacity: 0, x: 20},
+              animate: {opacity: 1, x: 0},
+              exit: {opacity: 0},
+            }}
+          >
+            <PlatformIcon size={48} format="lg" platform={currentPlatform} />
+          </motion.div>
+        </TitleContainer>
```

Comment:
Can you extract this into it's own component?

Reply:
So it can be shared I mean, with the projectSetup thing

Reply:
updated